packages/snippets/bin/extract.dart

```diff
@@ -0,0 +1,142 @@
+/// to the new location.
+Future<void> main(List<String> argList) async {
+  final FlutterInformation flutterInformation = FlutterInformation();
+  final ArgParser parser = ArgParser();
+  parser.addOption(
+    _kOutputOption,
+    defaultsTo: path.join(flutterInformation.getFlutterRoot().absolute.path, 'examples', 'api'),
+    help: 'The output path for generated sample applications.',
+  );
+  parser.addOption(
+    _kInputOption,
+    mandatory: true,
+    help: 'The input Flutter source file containing the sample code to extract.',
+  );
+  parser.addFlag(
+    _kHelpOption,
+    defaultsTo: false,
+    negatable: false,
+    help: 'Prints help documentation for this command',
+  );
+
+  final ArgResults args = parser.parse(argList);
+
+  if (args[_kHelpOption] as bool) {
+    stderr.writeln(parser.usage);
+    exit(0);
+  }
+
+  if (args[_kInputOption] == null || (args[_kInputOption] as String).isEmpty) {
+    stderr.writeln(parser.usage);
+    errorExit('The --$_kInputOption option must not be empty.');
+  }
+
+  if (args[_kOutputOption] == null || (args[_kOutputOption] as String).isEmpty) {
+    stderr.writeln(parser.usage);
+    errorExit('The --$_kOutputOption option must be specified, and not empty.');
+  }
+
+  final File input = filesystem.file(args['input'] as String);
+  if (!input.existsSync()) {
+    errorExit('The input file ${input.absolute.path} does not exist.');
+  }
+  final String flutterSource = path.join(
+    flutterInformation.getFlutterRoot().absolute.path,
+    'packages',
+    'flutter',
+    'lib',
+    'src',
+  );
+  if (!path.isWithin(flutterSource, input.absolute.path)) {
+    errorExit('Input file must be under the $flutterSource directory: ${input.absolute.path} is not.');
+  }
+
+  final Iterable<SourceElement> fileElements = getFileElements(input);
+  final SnippetDartdocParser dartdocParser = SnippetDartdocParser();
+  final SnippetGenerator snippetGenerator = SnippetGenerator();
+  dartdocParser.parseFromComments(fileElements);
+  dartdocParser.parseAndAddAssumptions(fileElements, input, silent: true);
+
+  final String srcPath = path.relative(input.absolute.path, from: flutterSource);
+  final String dstPath = path.join(
+    flutterInformation.getFlutterRoot().absolute.path,
+    'examples',
+    'api',
+  );
+  for (final SourceElement element in fileElements.where((SourceElement element) {
+    return element.sampleCount > 0;
+  })) {
+    for (final CodeSample sample in element.samples) {
+      // Ignore anything else, because those are not full apps.
+      if (sample.type != 'dartpad' && sample.type != 'sample') {
+        continue;
+      }
+      final String relativePath = path.relative(sample.start.file!.path,
+          from: flutterInformation.getFlutterRoot().absolute.path);
+      snippetGenerator.generateCode(
+        sample,
+        includeAssumptions: false,
+        copyright: _kCopyrightNotice,
+        description: 'See description in the comments in the file:\n  $relativePath',
```

Comment:
Would the line number be helpful in the description here? There may be multiple samples in a file, right?

Reply:
No, I originally had that, but Michael pointed out that it wouldn't be a good idea because it'll get out of date as soon as the file changes.  Since the templates all have a comment that says "// Flutter code sample for {{element}}" in them, the actual result looks something like this:

```
// Flutter code sample for AboutListTile

// See description in the comments in the file:
//   packages/flutter/lib/src/material/about.dart
```

So they have a filename and a component name.  I could make it more explicit, with something like:
"See the description in the first sample listed for component AboutListTile of file packages/flutter/lib/src/material/about.dart", but I'm not sure it adds much.

Reply:
Ah that makes sense. What you have above sounds good.